WooCommerce/Classes/Authentication/AuthenticationManager.swift

```diff
@@ -457,9 +456,23 @@ private extension AuthenticationManager {
                              with credentials: AuthenticatorCredentials,
                              in navigationController: UINavigationController,
                              onDismiss: @escaping () -> Void) {
-        let viewModel = JetpackErrorViewModel(siteURL: siteURL, onJetpackSetupCompletion: { [weak self] in
-            self?.currentSelfHostedSite = nil
-            self?.presentLoginEpilogue(in: navigationController, for: credentials, onDismiss: onDismiss)
+        let viewModel = JetpackErrorViewModel(siteURL: siteURL, onJetpackSetupCompletion: { [weak self] authorizedEmailAddress in
+            guard let self = self else { return }
+            // Resets the referenced site since the setup completed now.
+            self.currentSelfHostedSite = nil
+            guard credentials.wpcom != nil else {
+                return WordPressAuthenticator.showLoginForJustWPCom(
+                    from: navigationController,
+                    jetpackLogin: true,
+                    connectedEmail: authorizedEmailAddress,
+                    siteURL: siteURL
+                )
+            }
+            // Tries re-syncing to get an updated store list,
+            // then attempts to present epilogue again.
+            ServiceLocator.stores.synchronizeEntities { [weak self] in
+                self?.presentLoginEpilogue(in: navigationController, for: credentials, onDismiss: onDismiss)
```

Comment:
super nit: do you think it's still worth mentioning the condition that breaks the recursion, a site URL that has a valid Jetpack?

Reply:
I don't think it'll cause a recursion here since we reset `currentSelfHostedSite` to nil above. Does that make sense?

Reply:
oh I thought it was `isJetpackInvalidForSelfHostedSite` that breaks the recursion, but I guess it's mostly from `currentSelfHostedSite` being nil from `isJetpackInvalidForSelfHostedSite` implementation. Just thought it's nice to mention for future readers, but totally a non-blocking nit.